openedx/core/djangoapps/persistenttask/models.py

```diff
@@ -0,0 +1,32 @@
+"""
+Models to support persistent tasks.
+"""
+
+from django.db import models
+from jsonfield import JSONField
+
+
+class FailedTask(models.Model):
+    """
+    Representation of tasks that have failed
+    """
+    task_name = models.CharField(max_length=255)
+    task_id = models.CharField(max_length=255, db_index=True)
+    args = JSONField(blank=True)
+    kwargs = JSONField(blank=True)
+    exc = models.CharField(max_length=255)
+    datetime_failed = models.DateTimeField(db_index=True)
+    datetime_resolved = models.DateTimeField(blank=True, null=True, default=None)
+
+    class Meta(object):
+        index_together = [
+            (u'task_name', u'exc'),
```

Comment:
This provides the infrastructure for rerunning a particular type of task.  Ordinarily, I suspect the "task_name" part will be used more often by itself, but having the exception available as well allows us to rerun tasks that failed for a particular reason without rerunning ones that we know will just fail again.